File: bench/fib.py
# The Python twin of fib.sl.
#
# CPython's default recursion limit is 1000 frames and this goes 33 deep, so nothing has to be
# raised -- what the benchmark measures is the cost of a frame, not the ceiling on how many.


def fib(n):
    return n if n < 2 else fib(n - 1) + fib(n - 2)


print(fib(33))

File: bench/fib.rb
# The Ruby twin of fib.sl. Recursion 33 deep is nowhere near Ruby's stack limit, so what this
# measures is the cost of a method frame.

def fib(n)
  n < 2 ? n : fib(n - 1) + fib(n - 2)
end

puts fib(33)
